core/trino-main/src/test/java/io/trino/sql/query/QueryAssertions.java

```diff
@@ -323,10 +324,15 @@ private QueryAssert(
             this.skipResultsCorrectnessCheckForPushdown = skipResultsCorrectnessCheckForPushdown;
         }
 
-        // TODO for better readability, replace this with `exceptColumns(String... columnNamesToExclude)` leveraging MaterializedResult.getColumnNames
-        @Deprecated
-        public QueryAssert projected(int... columns)
+        public QueryAssert exceptColumns(String... columnNamesToExclude)
         {
+            List<String> columnNames = actual.getColumnNames();
+            Predicate<String> selectedColumns = ((Predicate<String>) Arrays.asList(columnNamesToExclude)::contains).negate();
```

Comment:
Add a test in `BaseQueryAssertionsTest` verifying that excluding a column that doesn't exist fails.